### PR TITLE
Add reno ci

### LINKED ISSUE #328: Add release notes to the openQASM Project
### What is the expected enhancement?

The openQASM project should have release notes associated with each tagged release (see #321). These should contain a summary of all changes included in the prepared release (or the development branch). For ease (and distribution) of maintenance, we should consider using [reno](https://docs.openstack.org/reno/latest/) which has been used with success in Qiskit Terra [1](https://github.com/Qiskit/qiskit-terra#changelog-and-release-notes), [2](https://github.com/Qiskit/qiskit-terra/blob/main/CONTRIBUTING.md#release-notes).

### PR DESCRIPTION
**Pull request by taalexander**
### Summary

Are you changing the specification? This PR needs to be approved by the TSC members.
No

Are you changing the grammar to adjust to the specification?
No

### Details and comments
Closes #328

### CODE REVIEW

**Comment by taalexander** [comment]
The test is to make sure it fails CI until I add a reno.

**Comment by taalexander** [comment]
Confirmed this is behaving as expected and ready for review:
```
ouput: Reno added or updated: releasenotes/notes/add-releasenote-checks-78194cf9de8ded3e.yaml
```

File: .github/workflows/validate-releasenotes.yml [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+          python-version: '3.10.9'
+      - name: Install pip packages
+        run: |
+          python -m pip install --upgrade pip
+          pip install -r requirements.txt
+      - name: Validate releasenotes
+        if: github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'no-reno')
+        env:
+          BRANCH_NAME: ${{ github.head_ref || github.ref_name }}
+        run: ./.github/renos_updated.sh
```
Comment (hodgestar) [inline .github/workflows/validate-releasenotes.yml:28]:
```suggestion
      - name: Lint releasenotes
        run: reno lint
      - name: Check releasenotes updated
        if: github.event_name == 'pull_request' && !contains(github.event.pull_request.labels.*.name, 'no-reno')
        run: ./.github/renos_updated.sh
```
It seems a bit clearer to me to always lint the release notes and then only check if something was added in pull requests without the `no-reno` label. I also couldn't find the `BRANCH_NAME` used anywhere.

File: .github/renos_updated.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+# This script makes sure a reno has been updated in the PR.
+
+reno lint
+
```
Comment (hodgestar) [inline .github/renos_updated.sh:5]:
```suggestion
```
See suggestion to always run `reno lint` and just put it directly in the job steps.

**Review by hodgestar — CHANGES_REQUESTED** [review]
Looks good to me.

I suggested a few things that seemed like small improvements to me.

When I run `reno report` it generates a report for `ast-py/v0.5.0-19` which is rather odd. Partially this is because of the old tag names in the repository, but I'm not sure what the `-19` is about. Can we do anything about this in this PR? Or do we just need to fix things when we sort out the release process itself?

File: .github/workflows/validate-releasenotes.yml [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+    types: [opened, reopened, labeled, unlabeled, synchronize]
+jobs:
+  Build:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+        with:
+          fetch-depth: 0
+      - name: Set up Python
+        uses: actions/setup-python@v4
```
[the remark this answers is not in the document]
Comment (taalexander) [reply .github/workflows/validate-releasenotes.yml:17]: Addressed in 9b7834d

File: .github/workflows/validate-releasenotes.yml [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+        with:
+          python-version: '3.10.9'
```
[the remark this answers is not in the document]
Comment (taalexander) [reply .github/workflows/validate-releasenotes.yml:19]: Addressed in 9b7834d

File: .github/renos_updated.sh (hunk 2 above)
Comment (taalexander) [reply .github/renos_updated.sh:5]: Completed in 737697a

File: .github/workflows/validate-releasenotes.yml (hunk 1 above)
Comment (taalexander) [reply .github/workflows/validate-releasenotes.yml:28]: Completed in 737697a

**Comment by taalexander** [comment]
Reno report looks ok for me.
```
=============
Release Notes
=============

.. _Release Notes_3.0-225:

3.0-225
=======

.. _Release Notes_3.0-225_New Features:

New Features
------------

.. releasenotes/notes/add-releasenote-checks-78194cf9de8ded3e.yaml @ b'c44c713ed42eafbd7d9d87340ec41891113cbc37'

- Added CI support for ensuring contributions include a release note.
  If this is not desired for a PR the tag "no-reno" may be added to the
  PR.


.. _Release Notes_3.0-225_Upgrade Notes:

Upgrade Notes
-------------

.. releasenotes/notes/clarify-physical-qubits-f720f21a3d792c70.yaml @ b'782c3b6986e5ff46757a8867c39bf6f9acac603b'

- Improve physical qubit documentation. Define what constitutes a valid
  physical identifier, define physical circuits, and answer several questions
  about what a compiler may do with physical qubits.

.. releasenotes/notes/drop-dollar-from-pulse-wildcard-2efa545932ed7736.yaml @ b'ce7548adcf1b34941df112f8ecb1194726fd9acd'

- The "wildcard" identifiers in pulse grammars are now regular OpenQASM 3
  identifiers, without the leading dollar sign (``$``), *e.g.* what was ``$q``
  before is now simply ``q``.  Precise hardware qubits are still referenced
  as ``$0``, for example.  This was made for better consistency in identifiers;
  having the dollar sign in the wildcard carried no information, since it
  applied to any qubit already.
```

**Review by hodgestar — APPROVED** [review]

**Comment by taalexander** [comment]
I just ran it with `reno report` full output
```
(qss-compiler) thomas@Thomass-MacBook-Pro openqasm-1 % reno report                            
loaded configuration file ./releasenotes/config.yaml
scanning ./releasenotes/notes (branch=*current* earliest_version=None collapse_pre_releases=True)
including entire branch history
78194cf9de8ded3e: update to releasenotes/notes/add-releasenote-checks-78194cf9de8ded3e.yaml in commit b'c44c713ed42eafbd7d9d87340ec41891113cbc37'
78194cf9de8ded3e: adding releasenotes/notes/add-releasenote-checks-78194cf9de8ded3e.yaml from 3.0-225
78194cf9de8ded3e: copying data for releasenotes/notes/add-releasenote-checks-78194cf9de8ded3e.yaml from commit b'c44c713ed42eafbd7d9d87340ec41891113cbc37'
f720f21a3d792c70: adding releasenotes/notes/clarify-physical-qubits-f720f21a3d792c70.yaml from 3.0-225
2efa545932ed7736: adding releasenotes/notes/drop-dollar-from-pulse-wildcard-2efa545932ed7736.yaml from 3.0-225
001094 b'7fbea5f0a7e43f1b46b2c739b700c3b9d56c5146' updating current version to 3.0
=============
Release Notes
=============

.. _Release Notes_3.0-225:

3.0-225
=======

.. _Release Notes_3.0-225_New Features:

New Features
------------

.. releasenotes/notes/add-releasenote-checks-78194cf9de8ded3e.yaml @ b'c44c713ed42eafbd7d9d87340ec41891113cbc37'

- Added CI support for ensuring contributions include a release note.
  If this is not desired for a PR the tag "no-reno" may be added to the
  PR.


.. _Release Notes_3.0-225_Upgrade Notes:

Upgrade Notes
-------------

.. releasenotes/notes/clarify-physical-qubits-f720f21a3d792c70.yaml @ b'782c3b6986e5ff46757a8867c39bf6f9acac603b'

- Improve physical qubit documentation. Define what constitutes a valid
  physical identifier, define physical circuits, and answer several questions
  about what a compiler may do with physical qubits.

.. releasenotes/notes/drop-dollar-from-pulse-wildcard-2efa545932ed7736.yaml @ b'ce7548adcf1b34941df112f8ecb1194726fd9acd'

- The "wildcard" identifiers in pulse grammars are now regular OpenQASM 3
  identifiers, without the leading dollar sign (``$``), *e.g.* what was ``$q``
  before is now simply ``q``.  Precise hardware qubits are still referenced
  as ``$0``, for example.  This was made for better consistency in identifiers;
  having the dollar sign in the wildcard carried no information, since it
  applied to any qubit already.

```

**Comment by taalexander** [comment]
I'll need another approving review to merge it seems.

**Review by hodgestar — APPROVED** [review]

**Comment by braised-babbage** [comment]
> Looks good to me.
> 
> I suggested a few things that seemed like small improvements to me.
> 
> When I run `reno report` it generates a report for `ast-py/v0.5.0-19` which is rather odd. Partially this is because of the old tag names in the repository, but I'm not sure what the `-19` is about. Can we do anything about this in this PR? Or do we just need to fix things when we sort out the release process itself?

I saw this as well. Just tinkering with things, I noticed that when I added a commit, the number got incremented. When I added a dummy `ast-py/v0.5.1` tag, the report used this without any suffix. So my guess is that `reno report` will include notes that have occurred since the last tagged release, indicating the number of commits since then. The recent release notes were added after the `ast-py/v0.5.0` release.

**Review by braised-babbage — APPROVED** [review]
Thanks for adding this.

**Pull request merged**